File: macOS/MediaFlowMonitor/Sources/MediaFlowMonitor/Overlay/MetricsHistory.swift
import Foundation

/// Un punct de istoric pentru grafice — timestamp + valoare.
struct HistoryPoint: Identifiable {
    let id = UUID()
    let date: Date
    let value: Double
}

/// Ring-buffer simplu pentru seriile de timp afișate în grafice
/// (VRAM/Swap utilization, CPU per-thread) — capacitate fixă, nu crește
/// nemărginit cât timp aplicația rulează.
final class MetricsHistory {
    private let capacity: Int
    private(set) var points: [HistoryPoint] = []

    init(capacity: Int = 120) { // ~ 6 minute la 3s/tick
        self.capacity = capacity
    }

    func append(_ value: Double, at date: Date = Date()) {
        points.append(HistoryPoint(date: date, value: value))
        if points.count > capacity {
            points.removeFirst(points.count - capacity)
        }
    }
}

File: Windows/MediaFlowMonitor/Overlay/MetricsHistory.cs
using System;
using System.Collections.Generic;

namespace MediaFlowMonitor.Overlay;

public readonly record struct HistoryPoint(DateTime Date, double Value);

/// Ring-buffer pentru seriile de timp din grafice — echivalentul
/// MetricsHistory.swift de pe Mac, capacitate fixă (nu crește nemărginit).
public sealed class MetricsHistory
{
    private readonly int _capacity;
    private readonly List<HistoryPoint> _points = new();

    public IReadOnlyList<HistoryPoint> Points => _points;

    public MetricsHistory(int capacity = 120) => _capacity = capacity;

    public void Append(double value, DateTime? at = null)
    {
        _points.Add(new HistoryPoint(at ?? DateTime.Now, value));
        if (_points.Count > _capacity)
            _points.RemoveRange(0, _points.Count - _capacity);
    }
}
